homeassistant/components/alexa/__init__.py

```diff
@@ -49,6 +50,7 @@
         vol.Optional(CONF_LOCALE, default=DEFAULT_LOCALE): vol.In(
             CONF_SUPPORTED_LOCALES
         ),
+        vol.Optional(CONF_HASS_URL): cv.url,
```

Comment:
Why can't we use `helpers.async_get_external_url(hass)` ?

Reply:
`helpers.async_get_external_url(hass)` is called in the `AlexaConfig` class if this value isn't set.

Reply:
In what case would we want the user to override this ? Doesn't that just mean that they should configure their base url instead ?

Reply:
It's likely a rare case, but it's something I had to implement to work in my network since the request is coming directly from the echo device on the same LAN but requires a common trusted certificate. 

My base URL is an internal intranet URL and roll my own Certificate Authority for all local devices that support certs. External requests are proxied through an edge instance with a public facing domain and a cert from a Common Trusted CA. 

Therefore my base URL wouldn't have a trusted cert, since I can't add my root CA to Alexa devices, and while I do subscribe to HA cloud to support HA, I choose to roll my Alexa Skill, and maintain remote access through my own VPN. :sweat_smile:

Reply:
So I would like to solve this once and for all, instead of having to solve this also for other integrations in the future.

So instead of this, maybe we should introduce a new 2nd configuration option for core that besides a base url, you can also set a base_url_internal? And then in your case, another option for `cloud` to not use the external url ?

Reply:
Why do you have your base url set to an internal url with a custom cert? That is not something that can be shared with other devices. Shouldn't that your public facing domain ?

Reply:
I've been discussing this with @frenck and we explored different strategies, but in the end we realized that it works as expected, as long as your base url is set up for external facing, hence my question. It seems to be very specific to your situation so I would like to understand it better.

Reply:
My situation and implementation steams from a couple things; my netsec background, and my uncanny ability to make things as difficult as possible. 😉 

After you proposed adding an internal base URL it caused me to step back and think about this situation and what problem I was solving. I'm embarrassed to admit I didn't realize this until now, I was trying to implement a work-around... for a development work-around I created months ago to develop and test this capability, but had forgotten why. 🤦‍♂ At the time I didn't want to reconfigure my HA instance, and my Lambda function to use a different cert, so I proxied these connections on another domain with a LetsEncrypt cert. 

If a user wants to make use of this implementation they will need to have a public facing domain with a cert from an authority trusted by Amazon. And 99% of users are going to satisfy those requirements with either HA Cloud, or LetsEncrypt/DuckDNS.

Thanks for setting me straight. 👍 I'll make the changes to remove the custom URL configuration.

Reply:
Changes have been pushed.